# MODELO SIR: Ajuste de datos para México

José Domingo Cajina Ramírez \\
Proyecto Final: Investigación de Operaciones


---

Este es uno de los modelos epidemiológicos mas sencillos capaz de describir con pocas suposiciones un brote epidémico.

 El nombre del modelo viene de las iniciales S (Susceptible), I (Infectados) y R ( Recuperados), donde S son todas las personas que podrían contraer la enfermedad, I son la cantidad de infectados iniciales en el sistema y R son las personas que han adquirido inmunidad, ya se por superar la enfermedad o por haber fallecido.



---

Se desea realizar un ajuste del modelo a la evolución de la pandemia en México, por lo que descargamos la base de datos de un sitio oficial y la incluimos en el repositorio de este notebook, comenzamos prepocesando los datos para dejarlos listos para el análisis



In [0]:
#importamos librerias
import pandas as pd
import matplotlib.pyplot as plt


In [0]:
#Descargamos Data del repositorio

#Datos de casos confirmados
!wget -cq https://raw.githubusercontent.com/Physicworld/ProyectoIO/master/Data_Base/Casos_Diarios_Estado_Nacional_Confirmados.csv

#Datos de defunciones
!wget -cq https://raw.githubusercontent.com/Physicworld/ProyectoIO/master/Data_Base/Casos_Diarios_Estado_Nacional_Defunciones.csv

#Datos de casos negativos
!wget -cq https://raw.githubusercontent.com/Physicworld/ProyectoIO/master/Data_Base/Casos_Diarios_Estado_Nacional_Negativos.csv

#Datos de casos sospechosos
!wget -cq https://raw.githubusercontent.com/Physicworld/ProyectoIO/master/Data_Base/Casos_Diarios_Estado_Nacional_Sospechosos.csv

In [0]:
#Creamos un dataframe de los datos

df = pd.read_csv("Casos_Diarios_Estado_Nacional_Confirmados.csv", encoding='Latin-1')
data_casos_confirmados = pd.DataFrame(df)

df = pd.read_csv("Casos_Diarios_Estado_Nacional_Defunciones.csv", encoding='Latin-1')
data_casos_defunciones = pd.DataFrame(df)

df = pd.read_csv("Casos_Diarios_Estado_Nacional_Negativos.csv", encoding='Latin-1')
data_casos_negativos = pd.DataFrame(df)

df = pd.read_csv("Casos_Diarios_Estado_Nacional_Sospechosos.csv", encoding='Latin-1')
data_casos_sospechosos = pd.DataFrame(df)


# Damos pantalla a cada DataFrame

In [62]:
data_casos_confirmados.head(3)

cve_ent  poblacion           nombre  ...  07-05-2020  08-05-2020  09-05-2020
0        0  127792286         Nacional  ...         716         356          44
1        1    1434635   AGUASCALIENTES  ...           0           0           0
2        2    3634868  BAJA CALIFORNIA  ...          19           3           2

[3 rows x 127 columns]

In [63]:
data_casos_defunciones.head(3)

cve_ent  poblacion           nombre  ...  08-05-2020  09-05-2020  10-05-2020
0        0  127792286         Nacional  ...          55          20           3
1        1    1434635   AGUASCALIENTES  ...           0           0           0
2        2    3634868  BAJA CALIFORNIA  ...          13           1           0

[3 rows x 58 columns]

In [64]:
data_casos_negativos.head(3)

cve_ent  poblacion           nombre  ...  07-05-2020  08-05-2020  09-05-2020
0        0  127792286         Nacional  ...        1271         578          87
1        1    1434635   AGUASCALIENTES  ...           2           1           0
2        2    3634868  BAJA CALIFORNIA  ...           9           1           0

[3 rows x 134 columns]

In [65]:
data_casos_sospechosos.head(3)

cve_ent  poblacion           nombre  ...  08-05-2020  09-05-2020  10-05-2020
0        0  127792286         Nacional  ...        4735        2402         640
1        1    1434635   AGUASCALIENTES  ...          44          14           1
2        2    3634868  BAJA CALIFORNIA  ...         140          59          18

[3 rows x 52 columns]

Nuestro objeto de estudio es el pais entero y no particularmente cada estado, podemos ver que cada grupo de datos nos da los datos nacionales en cada fecha, por lo cual nos quedaremos únicamente con la primera fila de cada dataframe.


In [74]:
df = data_casos_confirmados.drop(range(1,len(data_casos_confirmados)), axis=0)
data_casos_confirmados = df
data_casos_confirmados

cve_ent  poblacion    nombre  ...  07-05-2020  08-05-2020  09-05-2020
0        0  127792286  Nacional  ...         716         356          44

[1 rows x 127 columns]

In [75]:
df = data_casos_defunciones.drop(range(1,len(data_casos_defunciones)), axis=0)
data_casos_defunciones = df
data_casos_defunciones

cve_ent  poblacion    nombre  ...  08-05-2020  09-05-2020  10-05-2020
0        0  127792286  Nacional  ...          55          20           3

[1 rows x 58 columns]

In [76]:
df = data_casos_negativos.drop(range(1,len(data_casos_negativos)), axis=0)
data_casos_negativos = df
data_casos_negativos

cve_ent  poblacion    nombre  ...  07-05-2020  08-05-2020  09-05-2020
0        0  127792286  Nacional  ...        1271         578          87

[1 rows x 134 columns]

In [77]:
df = data_casos_sospechosos.drop(range(1,len(data_casos_sospechosos)), axis=0)
data_casos_sospechosos = df
data_casos_sospechosos

cve_ent  poblacion    nombre  ...  08-05-2020  09-05-2020  10-05-2020
0        0  127792286  Nacional  ...        4735        2402         640

[1 rows x 52 columns]

Un dato en comun de cada data frame es la poblacion nacional podemos ver que es de 127792286 	habitantes.

La intencion es crear una sola tabla de datos para toda la estadísitica para su fácil manejo, por que lo que ponemos de index las fechas, transponemos cada dataframe y los unimos.

In [0]:
casos_mexico = pd.concat([data_casos_confirmados,data_casos_negativos, data_casos_sospechosos, data_casos_defunciones], axis=0)
casos_mexico = casos_mexico.fillna(0)

In [85]:
casos_mexico

cve_ent  poblacion    nombre  ...  05-01-2020  06-01-2020  10-05-2020
0        0  127792286  Nacional  ...         0.0         0.0         0.0
0        0  127792286  Nacional  ...        14.0        28.0         0.0
0        0  127792286  Nacional  ...         0.0         0.0       640.0
0        0  127792286  Nacional  ...         0.0         0.0         3.0

[4 rows x 135 columns]